apps/src/code-studio/components/progress/progress_dot.jsx

```diff
@@ -185,7 +187,8 @@ export const ProgressDot = React.createClass({
     const showLevelName = /(named_level|peer_review)/.test(level.kind) && this.props.courseOverviewPage;
     const isPeerReview = level.kind === 'peer_review';
     const iconForLevelStatus = !isUnplugged && this.props.courseOverviewPage && this.getIconForLevelStatus(level);
-    const levelUrl = level.locked ? undefined : level.url + location.search;
+    const isLocked = level.locked || level.status === LevelStatus.locked;
```

Comment:
Could have sworn I responded to this comment already, but perhaps it didn't go through because of the internet.

`level.locked` is a per level user-agnostic concept, I believe only used by peer reviews ( @mehalshah can probably confirm), which I believe we find out about as part of page load in `scriptData` . `level.status === LevelStatus.locked` is a user-specific concept, which we don't know about until we request user progress.

Not sure if there's a better way of disambiguating these two concepts.